### ISSUE TITLE
oosmos_TimeoutHasExpired suboptimal

### ISSUE THREAD

**Issue by dcurrie** [oosmos/oosmos#1]
`oosmos_TimeoutHasExpired` could be much simpler, a one liner!,  if `Now` and `End` were kept as `uint32_t`:

```
    return (Now - Start) > (End - Start);
```

In fact, it would be simpler still if instead of recording `End` you recorded `Timeout` in `uint32_t` micros, then:

```
    return (Now - Start) > Timeout;
```

Unsigned arithmetic really works!

**Comment by oosmos** [comment]
An excellent and welcome improvement.  Has been included in version 2.4.0.

All improvement suggestions are very much appreciated.

**Comment by dcurrie** [comment]
Looks much better!
